### PR TITLE
update queries to include both id from index

### PR DESCRIPTION
**Pull request by yongliu-mdsol**

### CODE REVIEW

File: src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs [hunk 1]
```diff
@@ -102,14 +102,14 @@ public int GetJobCount(string[] tags, string stateName = null)
                     $@")
 select top {maxTags} j.StateName AS [Key], count(*) AS [Value]
 from [{_options.SchemaName}].Job j with (nolock)
-inner join cte on cte.Id = j.Id 
-inner join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id
+inner join cte on cte.Id = j.Id
+inner join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id and j.id = s.jobId
```
Comment (yongliu-mdsol) [inline src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs:106]: Index on State table contains both Id and JobId.
Although this looks redundant, it reduces query cost from 950 to 0.5 with millions jobs in the table

File: src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs [hunk 2]
```diff
@@ -198,21 +197,21 @@ select count(*)
...
 $@"
   where (@stateName IS NULL OR LEN(@stateName) = 0 OR j.StateName=@stateName)
 )
 select j.*, s.Reason as StateReason, s.Data as StateData
 from [{_options.SchemaName}].Job j with (nolock)
-inner join cte on cte.Id = j.Id 
-left join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id
-where cte.row_num between @start and @end 
+inner join cte on cte.Id = j.Id
+inner join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id and j.id = s.jobId
```
Comment (yongliu-mdsol) [inline src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs:207]: Index on State table contains both Id and JobId.
Although this looks redundant, it reduces query cost from 950 to 0.5 with millions jobs in the table

File: src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs [hunk 3]
```diff
@@ -166,13 +166,12 @@ private int GetJobCount(DbConnection connection, string[] tags, string stateName
 )
 select count(*)
 from [{_options.SchemaName}].Job j with (nolock)
-inner join cte on cte.Id = j.Id 
-left join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id";
```
Comment (yongliu-mdsol) [inline src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs:170]: The query doesn't use any data from State table. 
We should remove it to improve performance

**Comment by yongliu-mdsol** [comment]
Before change: 
Using Clustered Scan
image
image

After the change: 
Using Clustered Index
image
image

**Comment by yongliu-mdsol** [comment]
@erwin-faceit can you please review and merge this fix?

**Comment by yongliu-mdsol** [comment]
@r-win this is to fix a performance issue when searching jobs with a tag

**Comment by erwin-faceit** [comment]
Excellent change. No idea why the build failed, but I don't think this is because of your change.

**Pull request merged**